### PR TITLE
FIP-2

### PR DESCRIPTION
**Pull request by Joeysantoro**
EDIT: added the TribeDripper

Adds three contracts:
* EthReserveStabilizer which allows for exchanging FEI for ETH at a usd conversion rate
* EthPCVDripper which can send in the ETH to the reserve stabilizer in batches over time
* TribeDripper sends TRIBE to the FeiRewardsDistributor to smoothen the front-loading of rewards

The FeiRewardsDistributor calculates the amount to release every week by taking the total balance of TRIBE and subtracting unreleased future rewards.
When a new TRIBE airdrop happens, it frontloads all the rewards for the weeks that have already passed into the first week.
For FIP-2 this would lead to 6x rewards in week 1 (1x base rewards + 100% boost x 5 drips) instead of 2x.

The TribeDripper drips in 47m, 31m, and 22m TRIBE each over the first 3 weeks to smoothen the distribution.

Week 1: 1x base rewards + 47% boost x 5 drips = ~3.35x
Week 2: 1.47x base rewards + 31% x 6 drips = ~3.33x
Week 3: 1.78x base rewards + 22% x 7 drips = ~3.32x
Week 4+: 2x base rewards

These would be voted on per FIP-2: 
https://tribe.fei.money/t/fip-2-fei-redemption-tribe-staking/3033

### CODE REVIEW

**Review by GNSPS — APPROVED** [review]

File: contracts/pcv/EthPCVDripper.sol [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+   using Address for address payable;
+ 
+   /// @notice target address to drip to
+   address public target;
+
+   /// @notice amount to drip after each window
+   uint256 public amountToDrip;
+
+   event Dripped(uint256 amount);
+   event Withdrawl(address indexed to, uint256 amount);
```
Comment (eswak) [inline contracts/pcv/EthPCVDripper.sol:19]: small typo here in the event name

File: contracts/pcv/IReserveStabilizer.sol [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    event Withdrawal(
+        address indexed _caller,
+        address indexed _to,
+        uint256 _amount
+    );
+
+    // ----------- State changing api -----------
+
+    function exchangeFeiForEth(uint256 feiAmount) external returns (uint256);
```
Comment (eswak) [inline contracts/pcv/IReserveStabilizer.sol:20]: in the interface you probably want more generic names (a `DaiReserverStabilizer` with an `exchangeFeiForEth` function would be weird)

File: contracts/pcv/EthPCVDripper.sol [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+   function drip()
+       external
+       afterTime
+       whenNotPaused
+   {
+       // reset timer
+       _initTimed();
+
+       // drip
+       payable(target).sendValue(amountToDrip);
```
Comment (Corsaire) [inline contracts/pcv/EthPCVDripper.sol:59]: Would be good to only make the transfer if the target’s balance is lower than some threshold.

File: contracts/pcv/EthPCVDripper.sol (hunk 1 above)
Comment (Joeysantoro) [reply contracts/pcv/EthPCVDripper.sol]: fixed

File: contracts/pcv/IReserveStabilizer.sol (hunk 2 above)
Comment (Joeysantoro) [reply contracts/pcv/IReserveStabilizer.sol]: renamed it to exchangeFei

File: contracts/pcv/EthPCVDripper.sol (hunk 1 above)
Comment (ruihop) [reply contracts/pcv/EthPCVDripper.sol]: nic

**Pull request merged**